### ISSUE TITLE
Member variables of forward-referenced class types are ignored during serialization

### ISSUE THREAD

**Issue by seri114** [Fatal1ty/mashumaro#70]
Member variables of forward-referenced class types are ignored during serialization.

For example, consider the following case:
```
from __future__ import annotations
from dataclasses import dataclass
from mashumaro import DataClassDictMixin

@dataclass
class A(DataClassDictMixin):
    a:B

@dataclass
class B(DataClassDictMixin):
    b:int

a = A(B(1))
print(a.to_dict())
```
This gives us the expected:
```
{'a': {'b': 1}}
```

On the other hand, running the following case does not give the expected results.
```
@dataclass
class Base(DataClassDictMixin):
    pass

@dataclass
class A1(Base):
    a:B1

@dataclass
class B1(Base):
    b:int


a = A1(B1(1))
print(a.to_dict())
```
Result:
```
{}
```
At this time, no exception or error message will be generated.

Expected:
```
{'a': {'b': 1}}
```

Thanks for the excellent tool!

**Comment by Fatal1ty** [comment]
Hi @seri114 

Thank you for spotting this! It is fixed in [this](https://github.com/Fatal1ty/mashumaro/tree/refactoring) branch. The next release will be very soon.

**Comment by Fatal1ty** [comment]
It it's fixed in [3.0](https://github.com/Fatal1ty/mashumaro/releases/tag/v3.0).

**Comment by seri114** [comment]
I have confirmed that this has been fixed in the v3.0 release.
Thanks again!
@Fatal1ty
